### PR TITLE
Add support of raw message type parsing to class DatabaseFeedManagerMessageProcessor

### LINKED ISSUE #20615: Implement Dynamic JSON Processing Based on 'Type' Field in database feed manager
# Description

This issue addresses the need to implement dynamic JSON processing based on the value of the 'type' field when receiving data through the subscription to the content updater. Depending on the 'type' field value, the system should execute different processing methods:

- When 'type' is 'offsets', the standard execution method should be applied.
- When 'type' is 'raw', the processing should be done as JSON streaming, where each line represents an element.

## User Stories

- As a developer, I want the system to identify the 'type' field in incoming JSON data and apply the appropriate processing method.
- As a user, I expect the system to handle 'offsets' data differently from 'raw' data, providing a more efficient and context-specific processing approach.

## Tasks

1. JSON 'Type' Detection:
   - Implement logic to detect the 'type' field in incoming JSON data.

2. 'offsets' Processing:
   - Ensure that 'offsets' data is handled appropriately.

3. 'Raw' JSON Streaming:
   - Implement JSON streaming processing for 'raw' data.
   - Ensure that each line of 'raw' data represents a separate element.

4. Testing and Validation:
   - Create test cases to validate the dynamic processing based on 'type'.
   - Verify that 'offsets' and 'raw' data are processed correctly.

## Definition of Done

- The system detects the 'type' field in incoming JSON data.
- 'offsets' data is processed using the standard method.
- 'raw' data is processed as JSON streaming with separate lines representing elements.
- Test cases confirm the correct processing of different data types.

### PR DESCRIPTION
**Pull request by matias-braida**
|Related issue|
|---|
|Closes #20615 |

## Description

This PR adds the capacity to dynamic JSON processing messages based on the value of the 'type' field. 
UTs were added and also database feed manager testtool was modified.

## Tests

image

- Compilation without warnings in every supported platform
  - [ ] Linux
- [ ] Manual tests
- [ ] UTs running ok

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp
Comment (pereyra-m) [inline src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp:1]: We should add tests for the new code in `src/wazuh_modules/vulnerability_scanner/tests/unit/eventDecoder_test.cpp`.

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Gj!

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp [hunk 1]
```diff
@@ -33,7 +33,32 @@ class EventDecoder final : public AbstractHandler<std::shared_ptr<EventContext>>
      */
     std::shared_ptr<EventContext> handleRequest(std::shared_ptr<EventContext> data) override
     {
-        if ("create" == data->resource.at("type"))
+        if ("raw_create" == data->resource.at("type"))
```
[the remark this answers is not in the document]
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp:36]: This behavior was modified. Now the EventDecoder is unchanged.

File: src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/file_raw1.json
[the remark this answers is not in the document]
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager/file_raw1.json:1]: This is the real extract of a raw file.

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp [hunk 2]
```diff
@@ -33,7 +33,32 @@ class EventDecoder final : public AbstractHandler<std::shared_ptr<EventContext>>
...
+            flatbuffers::Parser parser;
+
+            if (!parser.Parse(cve5_SCHEMA) || !parser.Parse(data->resource.at("payload").dump().c_str()))
+            {
+                throw std::runtime_error("Unable to parse payload: " + parser.error_);
+            }
+
+            try
+            {
+                data->cvesDatabase->delete_(data->resource.at("resource"));
```
[the remark this answers is not in the document]
Comment (matias-braida) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/eventDecoder.hpp:47]: This behavior was modified. Now the whole database is cleaned.

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Pull request merged**